lib/ulid/generate.rb

```diff
@@ -1,84 +1,35 @@
+# frozen_string_literal: true
+
 require 'securerandom'
 require 'ulid/constants'
 
 module ULID
-
   module Generate
     include Constants
 
     # returns the binary ULID as Base32 encoded string.
     def encode32
-      # Optimized unrolled loop ahead.
-      # From https://github.com/RobThree/NUlid
-      # via https://github.com/oklog/ulid/blob/c3c01856f7e43fa64133819126887124d5f05e39/ulid.go#L154
-
-      id = @bytes.bytes
-
-      output = +''
-
-      # Base32 encodes 5 bits of each byte of the input in each byte of the
-      # output. That means each input byte produces >1 output byte and some
-      # output bytes encode parts of multiple input bytes.
-      #
-      # The end result is a human / URL friendly output string that can be
-      # decoded into the original bytes.
+      (hi, lo) = bytes.unpack("Q>Q>")
+      value = (hi << 64) | lo
```

Comment:
Not a big deal, but you can avoid an array allocation with `unpack1`

```suggestion
      high = bytes.unpack1("Q>")
      low = bytes.unpack1("@8Q>")
      value = (high << 64) | low
```

Reply:
won't this require a second pass over the bytes?